cosmosResponse_test.go

```diff
@@ -197,3 +197,127 @@ func TestStatusCodeToDescription(t *testing.T) {
 	// THEN
 	assert.Contains(t, desc, "unknown")
 }
+
+func TestExtractRetryConditions(t *testing.T) {
```

Comment:
pls add "go.uber.org/goleak" to ensure that the main replayLoop has no go-routine leak